README.md

```diff
@@ -16,3 +16,203 @@ This repository contains an implementation of a SignalR server and a SignalR cli
 Protocol encoding in JSON is fully supported, and there is MessagePack support for basic types.
+
+- [Install](#install)
+- [Getting Started](#getting-started)
+    - [Server side](#server-side)
+        - [Implement the HubInterface](#implement-the-hubinterface)
+        - [Serve with http.ServeMux](#serve-with-httpservemux)
+    - [Client side](#client-side)
+        - [Grab copies of the signalr scripts](#grab-copies-of-the-signalr-scripts)
+        - [Use a HubConnection to connect to your server Hub](#use-a-hubconnection-to-connect-to-your-server-hub)
+
+## Install
+
+With a [correctly configured](https://golang.org/doc/install#testing) Go toolchain:
+
+```sh
+go get -u github.com/philippeseith/signalr
+```
+
+## Getting Started
+
+SignalR uses a `signalr.HubInterface` instance to anchor the connection on the server and a javascript `HubConnection` object to anchor the connection on the client.
+
+### Server side
+
+#### Implement the HubInterface
+
+The easiest way to implement the `signalr.HubInterface` in your project is to declare your own type and embed `signalr.Hub` which implements that interface and will take care of all the signalr plumbing. You can call your custom type anything you want so long as it implements the `signalr.HubInterface` interface.
+
+```go
+package main
+
+import "github.com/philippseith/signalr"
+
+type AppHub struct {
+    signalr.Hub
+}
+```
+
+Add functions with your custom hub type as a receiver.
+
+```go
+func (h *AppHub) SendChatMessage(message string) {
+    h.Clients().All().Send("chatMessageReceived", message)
+}
+```
+
+These functions must be public so that they can be seen by the signalr server package but can be invoked client-side as lowercase message names.  We'll explain setting up the client side in a moment, but as a preview, here's an example of calling our `AppHub.SendChatMessage(...)` method from the client:
+
+```js
+    // javascript snippet invoking that AppHub.Send method from the client
+    connection.invoke('sendChatMessage', val);
+```
+
+The `signalr.HubInterface` contains a pair of methods you can implement to handle connection and disconnection events.  `signalr.Hub` contains empty implementations of them to satisfy the interface, but you can "override" those defaults by implementing your own functions with your custom hub type as a receiver:
+
+```go
+func (c *chat) OnConnected(connectionID string) {
+    fmt.Printf("%s connected\n", connectionID)
+}
+
+func (c *chat) OnDisconnected(connectionID string) {
+   fmt.Printf("%s disconnected\n", connectionID)
+}
+```
+
+#### Serve with http.ServeMux
+
+```go
+import "net/http"
+
+func runHTTPServer() {
+    address := 'localhost:8080'
+    
+    // create an instance of your hub
+    hub := AppHub{}
+	
+    // build a signalr.Server using your hub
+    server, _ := signalr.NewServer(context.TODO(), signalr.SimpleHubFactory(hub))
+    
+    // configure global signalr options
```

Comment:
These options are not global, they need to be passed to NewServer(). See [Functional options for friendly APIs](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) for the pattern.

Reply:
thank you, I will review that.

Reply:
Ah, I see. I misread this in the `chatsample` `runHTTPServer` function. Will fix this up.

Reply:
fixed